test/e2e/knativeservingdeployment_test.go

```diff
@@ -1,7 +1,7 @@
 // +build e2e
 
```

Comment:
how is this deployment test different than the post upgrade test?

Reply:
The commonality is that at the beginning we need to create and make sure the operator CR is ready, and in the end remove the operator CR.

The difference is that integration tests tagged with `e2e` will run the verification of the knative serving deployments including configuration and recreation. The tests tagged with `postupgrade` will verify what should be ready after upgrading the operator from the previous release, like whether we have exactly the same number and names of deployments as expected(like in eventing operator, we have got a new deployment and removed an old one. The post upgrade tests there verify if we have the correct list of deployment). Later, I will move all the existing postupgrade tests available in knative serving repo into this post upgrade test in operator.

Reply:
Next step, I will open PRs to port the post upgrade tests in serving here to operator.